### PR TITLE
feat: implement narwhals.maybe_convert_dtypes

### LINKED ISSUE #241: Add `narwhals.maybe_convert_dtypes`
pandas-like libraries have `convert_dtypes`: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.convert_dtypes.html

People use this to "promote" types - e.g., to promote all types to use the pyarrow-backed ones.

Polars doesn't have anything like this, because each type is just self-standing. E.g. `pl.Int64` is just `pl.Int64`, there aren't any variants of it

So, the task here is to add a method `maybe_convert_dtypes`, which for pandas-like calls `convert_dtypes`, and for the rest is a no-op, just like how Narwhals already has other `maybe_*` funcs: https://narwhals-dev.github.io/narwhals/api-reference/narwhals/#narwhals.maybe_set_index

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #241 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/utils.py [hunk 1]
```diff
@@ -214,3 +214,85 @@ def maybe_set_index(df: T, column_names: str | list[str]) -> T:
             )
         )
     return df
+
+
+def maybe_convert_dtypes(  # noqa: PLR0913
+    df: T,
+    infer_objects: bool = True,  # noqa: FBT001, FBT002
```
Comment (MarcoGorelli) [inline narwhals/utils.py:221]:
```suggestion
    df: T,
    *,
    infer_objects: bool = True,  # noqa: FBT001, FBT002
```
(and then remove the noqas)

Comment (MarcoGorelli) [reply narwhals/utils.py:221]: though actually, all of these _might_ get renamed in pandas, and then we'd be stuck

Comment (DeaMariaLeon) [reply narwhals/utils.py]: Done.
I wonder why it doesn't fail locally.

File: tests/utils_test.py [hunk 2]
```diff
@@ -60,3 +60,20 @@ def test_maybe_set_index_polars() -> None:
...
     result = nw.maybe_set_index(df, "b")
     assert result is df
+
+
+def test_maybe_convert_dtypes_pandas() -> None:
+    import numpy as np
+
+    df = nw.from_native(pd.DataFrame({"a": [1.1, np.nan]}, dtype=np.dtype("float64")))
+    result = nw.to_native(nw.maybe_convert_dtypes(df))
+    expected = pd.DataFrame({"a": [1.1, pd.NA]}, dtype=pd.Float64Dtype())
```
Comment (MarcoGorelli) [inline tests/utils_test.py:70]: this fails on the minimum pandas version, could you use `Int64Dtype`?

Comment (DeaMariaLeon) [reply tests/utils_test.py]: Ups, I didn't see this comment before I tried the last commit. I'll update it.

**Comment by MarcoGorelli** [comment]
looks like pandas' series repr was a bit different in older versions (you can reproduce this locally by pip-installing the pandas version from the minimum-versions ci job) for string dtype

maybe only include bool and int in the example?

**Comment by DeaMariaLeon** [comment]
Thanks @MarcoGorelli

**Review by MarcoGorelli — APPROVED** [review]
thanks @DeaMariaLeon !

**Pull request merged**